src/dpi.rs

```diff
@@ -1,80 +1,106 @@
+//! them to think about growing elderly, resulting in them having an existential crisis. Once users
+//! enter that state, they will no longer be focused on your application.
 //!
-//! That's a description of what happens when the button is 100x100 *physical* pixels. Instead, let's try using 100x100
-//! *logical* pixels. To map logical pixels to physical pixels, we simply multiply by the DPI (dots per inch) factor.
-//! On a "typical" desktop display, the scale factor will be 1.0, so 100x100 logical pixels equates to 100x100 physical
-//! pixels. However, a 1440p display may have a scale factor of 1.25, so the button is rendered as 125x125 physical pixels.
-//! Ideally, the button now has approximately the same perceived size across varying displays.
+//! ## How should I handle it?
 //!
-//! Failure to account for the scale factor can create a badly degraded user experience. Most notably, it can make users
-//! feel like they have bad eyesight, which will potentially cause them to think about growing elderly, resulting in
-//! them entering an existential panic. Once users enter that state, they will no longer be focused on your application.
+//! The solution to this problem is to account for the device's *scale factor*. The scale factor is
+//! the factor UI elements should be scaled by to be consistent with the rest of the user's system -
+//! for example, a button that's normally 50 pixels across would be 100 pixels across on a device
+//! with a scale factor of `2.0`, or 75 pixels across with a scale factor of `1.5`.
 //!
-//! There are two ways to get the scale factor:
-//! - You can track the [`DpiChanged`](crate::event::WindowEvent::DpiChanged) event of your
-//!   windows. This event is sent any time the scale factor changes, either because the window moved to another monitor,
-//!   or because the user changed the configuration of their screen.
-//! - You can also retrieve the scale factor of a monitor by calling
-//!   [`MonitorHandle::scale_factor`](crate::monitor::MonitorHandle::scale_factor), or the
-//!   current scale factor applied to a window by calling
-//!   [`Window::scale_factor`](crate::window::Window::scale_factor), which is roughly equivalent
-//!   to `window.current_monitor().scale_factor()`.
+//! The scale factor correlates with, but no has direct relationship to, the screen's actual DPI
+//! (dots per inch). Operating systems used to define the scale factor in terms of the screen's
+//! approximate DPI (at the time, 72 pixels per inch), but [Microsoft decided to report that the DPI
+//! was roughly 1/3 bigger than the screen's actual DPI (so, 96 pixels per inch) in order to make
+//! text more legible][microsoft_dpi]. As a result, the exact DPI as defined by the OS doesn't carry
+//! a whole lot of weight when designing cross-platform UIs. Scaled pixels should generally be used
+//! as the base unit for on-screen UI measurement, instead of DPI-dependent units such as
+//! [points][points] or [picas][picas].
```

Comment:
This is still confusing. winit doesn't expose points or picas so it's not clear why they're being discussed, and "correlates with, but has no direct relationship to" is a contradiction in terms.

Reply:
I chose to talk about points and picas because, before going through this review process, I thought they were the units you were supposed to use for font size, and I don't think that's an uncommon misconception to have.

My thoughts when writing that phrase was `correlates = relates to, though not necessarily directly` and `direct relationship = there is a clear mathematical rule you can use to derive property B from property A`. I'll admit that that's a somewhat lawerly application or language though.